mass_mailing_event_registration_exclude/models/event_registration.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2016 Tenativa - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class EventRegistration(models.Model):
+    _inherit = "event.registration"
+
+    @api.model
+    def search_count(self, domain, limit=None):
+        res = super().search_count(domain, limit=limit)
+        mass_mailing_id = self.env.context.get("exclude_mass_mailing", False)
+        if mass_mailing_id:
+            res_ids = self.env["mailing.mailing"].event_filtered_ids(
+                self, mass_mailing_id, domain, field="email"
+            )
+            res = len(res_ids) if res_ids else 0
+        return res
```

Comment:
I think this could be “moved” to an AbstractModel (I'm not sure what the name could be) to avoid repeating this piece of code in several models: `event.registration`, `mailing.contact`, and `res.partner` (it will only be necessary to inherit the AbstractModel in those models). What do you think?